src/server/service/user.ts

```diff
@@ -31,7 +30,7 @@ export default class UserService implements IUserService {
   }
 
   async create(dto: CreateUserDTO): Promise<User> {
-    const user = new UserModel(dto);
+    const user = new UserModel({ role: dto.role ?? 'user', ...dto });
     const hash = await this.createPassword(dto.password);
 
     user.password = hash;
```

Comment:
As we are introducing roles, we should add some validation to the service when creating a user. We should make sure only users with a "higher" role are creating one user with a "lower" role.

What Im trying to say that `managers` should only be created by `admin` users and `users` should only be created by `manager` users.

Reply:
Here the `??` is assuming that you want to create a user with the `user` role if no role is specified in the DTO.
To make this more robust I suggest having `role` as a required property (remove the `?` from the type definition) and throwing an error when the `role` is not specified in the payload.

Creating a user with `user` role when no role is defined is unexpected behavior.

Reply:
> As we are introducing roles, we should add some validation to the service when creating a user. We should make sure only users with a "higher" role are creating one user with a "lower" role.
> 
> What Im trying to say that `managers` should only be created by `admin` users and `users` should only be created by `manager` users.

What are you suggesting? How to check if it is a "higher" role?